### DISCUSSION TITLE
Change password endpoint

### DISCUSSION THREAD

**Discussion by sinisaos** [piccolo-orm/piccolo_api#148] [Ideas]
I want to implement an endpoint to change the user password. Later we be able to use the same endpoint to change the user password in Piccolo Admin, which some people requested. Any thoughts on this will be welcome.

**Comment by dantownsend** [comment]
It's a good idea. Are you thinking of a form which people can use once they are logged in? Or a forgotten password form, which sends them a password reset link?

**Reply by sinisaos** [reply]
The form will only be for authenticated users. When the user is logged in she/he can change the password. This will be useful for Piccolo Admin (like Django admin change password). I already have a working solution and I can do PR so you can see and review the code. What do you think about that?

**Reply by dantownsend** [reply]
Yes, create a PR and I'll have a look.

It might be worth looking at best practices for password resets - e.g. should the user enter their current password before being able to set a new one.
